tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -923,3 +923,44 @@ def test_optuna_callback(self) -> None:
             tuner._tune_params(["num_leaves"], 10, optuna.samplers.TPESampler(), "num_leaves")
 
         assert callback_mock.call_count == 10
+
+    def test_get_best_booster(self) -> None:
+        unexpected_value = 20  # out of scope.
+
+        params = {"verbose": -1, "lambda_l1": unexpected_value}  # type: Dict
+        dataset = lgb.Dataset(np.zeros((10, 10)))
+
+        study = optuna.create_study()
+        with TemporaryDirectory() as tmpdir:
+            tuner = LightGBMTunerCV(
+                params,
+                dataset,
+                study=study,
+                model_dir=tmpdir,
+                return_cvbooster=True,
+            )
+            with mock.patch.object(_OptunaObjectiveCV, "_get_cv_scores", return_value=[1.0]):
+                tuner.tune_regularization_factors()
+
+            best_booster = tuner.get_best_booster()
+            for booster in best_booster.boosters:
+                assert booster.params["lambda_l1"] != unexpected_value
+
+        tuner2 = LightGBMTuner(params, dataset, valid_sets=dataset, study=study)
+
+        # Resumed study does not have the best booster.
+        with pytest.raises(ValueError):
+            tuner2.get_best_booster()
+
+        # the `model_dir` argument is None
+        with pytest.raises(ValueError) as excinfo:
+            tuner3 = LightGBMTunerCV(
+                params,
+                dataset,
+                study=study,
+                model_dir=None,
```

Comment:
I think the case where `model_dir` is `None` and `return_cvbooster=True` is missing. I'm not sure if we can skip it in the test. Do you think it is redundant?

Reply:
Apologies for the delay in response. I have added the case where `model_dir` is `None` and `return_cvbooster=True`.

I would like to get your advice for the opposite case where `model_dir` **is not** `None` and `return_cvbooster=False`.

Current behavior of `get_best_booster()` is as following:
1. if the best booster cannot be found in `model_dir`, `get_best_booster()` raise ValueError, which is expected.
2. if the best booster is found in `model_dir`, `get_best_booster()` still returns `lightgbm.CVBooster`.

Behavior 2 was bit counterintuitive to me, and I thought adding following check may simplify it.
Is this relating to your comment?
>BTW, I think we can get the best booster if we specify model_dir. What do you think?
```
        if self.lgbm_kwargs.get("return_cvbooster") is not True:
            raise ValueError(
                "LightGBMTunerCV requires `return_cvbooster=True` for method `get_best_booster()`."
            )
```